src/coord/src/catalog.rs

```diff
@@ -1866,6 +1866,41 @@ impl Catalog {
         }
     }
 
+    /// Reports whether the item identified by `id` is considered volatile.
+    ///
+    /// `None` indicates that the volatility of `id` is unknown.
+    pub fn is_volatile(&self, id: GlobalId) -> Option<bool> {
```

Comment:
Light preference to use an enum instead of nullable bool here, which I think makes a bad SQL interface due to its 2-but-actually-3 states. This would also change the SQL column type to either char or text, ala `provolatile` of https://www.postgresql.org/docs/current/catalog-pg-proc.html.

Reply:
Oof ok, I wish I'd thought of this. All fixed now.